src/Classes/ModStore.lua

```diff
@@ -501,8 +501,40 @@ function ModStoreClass:EvalMod(mod, cfg)
 				return
 			end
 		elseif tag.type == "SocketedIn" then
-			if not cfg or tag.slotName ~= cfg.slotName or (tag.keyword and (not cfg or not cfg.skillGem or not calcLib.gemIsType(cfg.skillGem, tag.keyword))) then
+			if not cfg or tag.slotName ~= cfg.slotName or (tag.socketColor ~= nil and (tag.sockets == nil or (tag.sockets ~= nil and type(tag.sockets) ~= "string")) and tag.socketColor ~= cfg.socketColor) or (tag.keyword and (not cfg or not cfg.skillGem or (not tag.neg and not calcLib.gemIsType(cfg.skillGem, tag.keyword) or tag.neg and calcLib.gemIsType(cfg.skillGem, tag.keyword)))) then
```

Comment:
I feel this could be handled with the match format which might simplify/generalise trying tag.neg checks. Some parts of this should probably return regardless e.g. if there is no cfg we should prob return.